vision.py

```diff
@@ -10,15 +11,72 @@
 from tqdm import tqdm
 from PIL import Image
 import tracking
+import time
+import moviepy.video.io.ImageSequenceClip
+from rq import get_current_job
+from flask_socketio import SocketIO
+from app import jsonify_data
 
-
-def get_tracking_video(fpath_video):
+def get_tracking_video(fpath_video, output_file):
+    '''
+    Given the path to a video, creates a tracking video and writes it to the
+    path output_file
+    Parameters:
+    - str fpath_video: path to video file
+    - str output_file: path to output file (doesn't need to already exist)
+    '''
+    print('Cuda available {}'.format(torch.cuda.is_available()))
+    job = get_current_job()
+    job_id = str(job.id)
+    pd = 'progress display'
+
+    # Opening socket
+    socketio = SocketIO(message_queue=os.getenv('REDIS_URL'), 
+                        cors_allowed_origins='*')
+    
+    # Getting images from video
+    status = 'Getting images from video'
+    socketio.emit(pd, {'status': status, 'id': job_id}, json=True)
+    job.meta['status'] = status
+    job.save()
+    print('Sent "Getting images from video" through socket')
+    video, extension = fpath_video.split('.')
     logger.info(f'get_tracking_video fpath_video: {fpath_video}')
     image_gen  = _get_images_from_video(fpath_video)
     images = [image for image in image_gen]
+    
+    # Getting predictions from images
+    status = 'Getting predictions from images'
+    socketio.emit(pd, {'status': status, 'id': job_id}, json=True)        
+    job.meta['status'] = status
+    job.save()
+    print('Sent "Getting getting predictions from images" through socket')  
     predictions = _get_predictions_from_images(images)
+    
+    # Getting tracks from predictions
+    status = 'Getting tracks from predictions'
+    socketio.emit(pd, {'status': status, 'id': job_id}, json=True)        
+    job.meta['status'] = status        
+    job.save()
```

Comment:
DRY: Don't Repeat Yourself

You've used this pattern several times:
```
    status = 'Getting tracks from predictions'
    socketio.emit(pd, {'status': status, 'id': job_id}, json=True)        
    job.meta['status'] = status        
    job.save()
```

Please move this to a function